### PR TITLE
Use Ingress v1 from Kubernetes v1.19 in helm chart

### PR DESCRIPTION
**Pull request by jlandowner**
Hi. I found you use old version of Ingress in helm chart.
Ingress became GA in [Kubernetes v1.19](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.19.md#ingress-graduates-to-general-availability).

Output after v1.19
```
# Source: code-server/templates/ingress.yaml
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: code-server
  labels:
    helm.sh/chart: code-server-1.0.3
    app.kubernetes.io/name: code-server
    app.kubernetes.io/instance: code-server
    app.kubernetes.io/version: "3.12.0"
    app.kubernetes.io/managed-by: Helm
spec:
  rules:
    - host: "code-server.example.loc"
      http:
        paths:
          - path: /
            backend:
              service:
                name: code-server
                port: 
                  number: 8080
```

### CODE REVIEW

**Comment by jsjoeio** [comment]
cc @kylecarbs @bpmct @jawnsy since I'm less familiar with this part of code-server

**Comment by bpmct** [comment]
This seems fine but am going to test with a cluster :)

Will add instructions here on how to spin up a cluster and try the helm chart :)

**Comment by jsjoeio** [comment]
> Will add instructions here on how to spin up a cluster and try the helm chart :)

You rock! Thank you so much!

**Comment by bpmct** [comment]
Actually, our existing [helm docs](https://github.com/cdr/code-server/blob/main/docs/helm.md) have lovely instructions on using the helm chart. Docker Desktop's built-in Kubernetes cluster or k3d is a good way to create a new local cluster.

I created a custom values.yaml to use Ingress:
```yaml
ingress:
  enabled: true
  annotations:
   kubernetes.io/ingress.class: nginx
  hosts:
   - host: 127.0.0.1.nip.io
     paths:
        - /
```

```sh
# if you haven't already, add ingress to cluster (https://kubernetes.github.io/ingress-nginx/deploy/)

git clone https://github.com/jlandowner/code-server
cd code-server
helm upgrade --install code-server ci/helm-chart --values values.yaml

# worked without deprecation notices! 
```

Just approved but will defer to @Matthew-Beckett since they maintain the chart :)

**Review by bpmct — APPROVED** [review]

**Comment by Matthew-Beckett** [comment]
@bpmct thanks Ben, will take a look!

**Review by Matthew-Beckett — APPROVED** [review]
Looks good to me! Thanks for your contribution ❤️

**Comment by bpmct** [comment]
@jsjoeio are we able to use CODEOWNERS or something so we can assign helm PRs to Matthew?

**Comment by jsjoeio** [comment]
> @jsjoeio are we able to use CODEOWNERS or something so we can assign helm PRs to Matthew?

Great idea! Yes, I'll do that now.

**Pull request merged**
